internal/ingest/ingest.go

```diff
@@ -716,16 +753,17 @@ func (ing *Ingester) ingestWorker() {
 		case <-ing.closeWorkers:
 			return
 		case provider := <-ing.toWorkers:
-			ing.ingestWorkerLogic(peer.ID(provider))
+			pid := peer.ID(provider)
+			pc := ing.providersBeingProcessed[pid]
+			pc <- struct{}{}
```

Comment:
Looks like there is still a race between when processing the ad chain finishes and the channel is cleared, [here](https://github.com/filecoin-project/storetheindex/commit/cf70a62e2ece09f18021fa940cb0c9fe2238d045#diff-50d2db719b550ec2183b831d2727d9184635a5647247ea8872ff82d91b6e31bdR432), and when the channel is again blocked for processing entries on the line above.

What happens if a new announce arrives after the channel is cleared, and before the channel is blocked above?   It looks execution would be blocked here waiting to write to the channel until the newly arrived announce was processed.  Could that cause a deadlock because then the `ing.toWorkers` channed is not being read?